### PR TITLE
[DOCS] Update routing formulas

### LINKED ISSUE #76072: The formula for calculating the shard which a document is routed in reference is wrong.
The formula for calculating the shard which a document is routed in reference is wrong. It does not match the actual calculation process in the code.

Formula in [reference](https://github.com/elastic/elasticsearch/blob/5baabff6670a8ed49297488ca8cac8ec12a2078d/docs/reference/mapping/fields/routing-field.asciidoc)
```
shard_num = hash(_routing) % num_primary_shards
```

Code 
```
indexMetaData.routingFactor = routingNumShards / numberOfShards;
Math.floorMod(hash, indexMetaData.getRoutingNumShards()) / indexMetaData.getRoutingFactor();
```

### PR DESCRIPTION
**Pull request by jrodewig**
The `_routing` metadata field docs currently include formulas for how Elasticsearch routes documents to shards. However,
these formulas were not updated for #18699.  This updates the routing formulas and adds xrefs for related settings.

Closes #76072

### Preview
https://elasticsearch_76203.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/mapping-routing-field.html

### CODE REVIEW

**Comment by jrodewig** [comment]
Hi @jtibshirani,

Let me know what you think of these changes. I can re-add the formulae if you feel they add information that's not adequately covered in these changes. Thanks!

**Comment by jrodewig** [comment]
I decided to just update the formulae here based on feedback in #76072. Thanks again!

File: docs/reference/mapping/fields/routing-field.asciidoc [hunk 1]
```diff
@@ -120,7 +125,7 @@ expense of having to search more shards per request.
 
 When this setting is present, the formula for calculating the shard becomes:
 
-    shard_num = (hash(_routing) + hash(_id) % routing_partition_size) % num_primary_shards
+    shard_num = (hash(_routing) + hash(_id) % routing_partition_size) % ( num_routing_shards / routing_factor )
```
Comment (jtibshirani) [inline docs/reference/mapping/fields/routing-field.asciidoc:128]: This also needs the parentheses to be adjusted:
```
shard_num = ((hash(_routing) + hash(_id) % routing_partition_size) % num_routing_shards) / routing_factor
```

The formula is actually getting pretty long, I had to stare at it a while to make sure I understood. I wonder if we could break it up into two:
```
routing_value = hash(_routing) + hash(_id) % routing_partition_size
shard_num = (routing_value %  num_routing_shards) / routing_factor
```

The second calculation matches the one above, which might make it easier to follow.

File: docs/reference/index-modules.asciidoc [hunk 2]
```diff
@@ -47,8 +47,10 @@ NOTE: The number of shards are limited to `1024` per index. This limitation is a
 `index.number_of_routing_shards`::
 +
 ====
-Number of routing shards used to <<indices-split-index,split>> an index.
+Integer value used with <<index-number-of-shards,`index.number_of_shards`>> to
+route documents to a primary shard. See <<mapping-routing-field>>.
 
+{es} also uses this value when <<indices-split-index,splitting>> an index.
```
Comment (jtibshirani) [inline docs/reference/index-modules.asciidoc:53]: Small comment, maybe we can remove the 'also' to strengthen this. My understanding is that the whole reason we have `index.number_of_routing_shards` is to make general splits/ shrinks possible, it doesn't really have a motivation on its own.

**Review by jtibshirani — COMMENTED** [review]
Nice detective work. I added a few comments about the formulas.

File: docs/reference/mapping/fields/routing-field.asciidoc [hunk 3]
```diff
@@ -2,12 +2,17 @@
 === `_routing` field
 
 A document is routed to a particular shard in an index using the following
-formula:
+formulas:
+    
+    routing_factor = num_routing_shards / num_primary_shards
+    shard_num = hash(_routing) % ( num_routing_shards / routing_factor )
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/mapping/fields/routing-field.asciidoc]: Thanks for catching this.

File: docs/reference/mapping/fields/routing-field.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/mapping/fields/routing-field.asciidoc]: Great suggestion.

File: docs/reference/index-modules.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/index-modules.asciidoc]: Good feedback. Thanks!

**Comment by jrodewig** [comment]
Thanks @jtibshirani! This is ready for another look at your convenience.

**Review by jtibshirani — APPROVED** [review]

**Comment by jrodewig** [comment]
Thanks @jtibshirani!

**Pull request merged**